### PR TITLE
Fix flaky upgrade test.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

The upgrade test would race with itself and occasionally fail.

Modifications:

Give enough time for all the work to settle before closing the
client channel.

Result:

No more flaky tests.

### CODE REVIEW

File: Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift [hunk 1]
```diff
@@ -679,7 +679,9 @@ class HTTPUpgradeTestCase: XCTestCase {
         try server.eventLoop.submit {
             upgrader.unblockUpgrade()
         }.wait()
-        client.close(promise: nil)
+        try client.eventLoop.scheduleTask(in: .milliseconds(10)) {
```
Comment (weissi) [inline Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift]: @Lukasa can you expand why it helps to wait 10ms here?

Comment (Lukasa) [reply Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift]: 10ms isn't a magic number, we just need to wait *long enough*.

The problem is that this close is racing with the `unblockUpgrade` operation above. It's possible that the close wins, and the connection is torn down before the client receives the 101 response. This wait tries to ensure that the client sees the 101 before connection termination.

Comment (Lukasa) [reply Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift]: We can do this another way by writing a custom accumulator handler that behaves differently to the one we're currently using, if you'd prefer.

Comment (weissi) [reply Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift]: @Lukasa sorry I think I'd prefer that 😬

Comment (Lukasa) [reply Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift]: Sure thing, I'll rework this.

**Comment by Lukasa** [comment]
Ok @weissi, updated.

**Comment by Lukasa** [comment]
test this please

**Pull request merged**